bin/src/command/orders.rs

```diff
@@ -357,93 +371,111 @@ impl CommandServer {
             Some(CommandResponseData::Workers(workers)),
         )
         .await;
+        Ok(())
     }
 
-    pub async fn launch_worker(&mut self, client_id: String, request_id: String, _tag: &str) {
-        if let Ok(mut worker) = start_worker(
+    pub async fn launch_worker(
+        &mut self,
+        client_id: String,
+        request_id: String,
+        _tag: &str,
+    ) -> anyhow::Result<()> {
+        let mut worker = match start_worker(
             self.next_id,
             &self.config,
             self.executable_path.clone(),
             &self.state,
             None,
         ) {
-            if let Some(sender) = self.clients.get_mut(&client_id) {
-                if let Err(e) = sender
-                    .send(CommandResponse::new(
-                        request_id.clone(),
-                        CommandStatus::Processing,
-                        "sending configuration orders".to_string(),
-                        None,
-                    ))
-                    .await
-                {
-                    error!("could not send message to client {:?}: {:?}", client_id, e);
-                }
+            Ok(worker) => worker,
+            Err(e) => {
+                error!("Failed at creating worker");
+                self.answer_error(client_id, request_id, "failed creating worker", None)
+                    .await;
+                return Err(e);
             }
+        };
 
-            info!("created new worker: {}", worker.id);
+        if let Some(sender) = self.clients.get_mut(&client_id) {
+            if let Err(e) = sender
+                .send(CommandResponse::new(
+                    request_id.clone(),
+                    CommandStatus::Processing,
+                    "sending configuration orders".to_string(),
+                    None,
+                ))
+                .await
+            {
+                error!("could not send message to client {:?}: {:?}", client_id, e);
+                // should we have an early return here?
```

Comment:
no early return here: the worker was started, so we'll support it in the main process (otherwise it might start serving traffic without being managed by the main process)

Reply:
okay thanks